src/Offer/Security/SecurityWithFacilityPermission.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace CultuurNet\UDB3\Offer\Security;
+
+use CultuurNet\UDB3\Offer\Commands\AuthorizableCommandInterface;
+use CultuurNet\UDB3\Offer\Security\Permission\PermissionVoterInterface;
+use CultuurNet\UDB3\Place\Commands\UpdateFacilities;
+use CultuurNet\UDB3\Security\SecurityDecoratorBase;
+use CultuurNet\UDB3\Security\SecurityInterface;
+use CultuurNet\UDB3\Security\UserIdentificationInterface;
+use ValueObjects\StringLiteral\StringLiteral;
+
+class SecurityWithFacilityPermission extends SecurityDecoratorBase
+{
+    /**
+     * @var UserIdentificationInterface
+     */
+    private $userIdentification;
+
+    /**
+     * @var PermissionVoterInterface
+     */
+    private $permissionVoter;
+
+    /**
+     * @param SecurityInterface $decoratee
+     * @param UserIdentificationInterface $userIdentification
+     * @param PermissionVoterInterface $permissionVoter
+     */
+    public function __construct(
+        SecurityInterface $decoratee,
+        UserIdentificationInterface $userIdentification,
+        PermissionVoterInterface $permissionVoter
+    ) {
+        parent::__construct($decoratee);
+
+        $this->userIdentification = $userIdentification;
+        $this->permissionVoter = $permissionVoter;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isAuthorized(AuthorizableCommandInterface $command)
+    {
+        //@todo: When extending for events create an interface.
+        // https://jira.uitdatabank.be/browse/III-2413
+        if ($command instanceof UpdateFacilities) {
```

Comment:
De naamgeving van deze class gaat al heel snel tot een groot aantal SecurityDecorators leiden, waarin je telkens wat code gaat moeten dupliceren. Bovendien lijkt ze eigenlijk niet veel te doen dat specifiek is voor de `VOORZIENINGEN_BEWERKEN` permissie, behalve dat je het binnenkomende command moet checken om te zien of je wel een permissie moet afchecken. Maar de permissie zelf haal je uit het command, en de permissionVoter is flexibel genoeg dat je eender welke permissie kan afchecken, dus die code is niet specifiek voor `VOORZIENINGEN_BEWERKEN`.

Kunnen we een manier zoeken dat we maar 1 decorator moeten voorzien, die dan op een bepaalde manier configureerbaar is om te bepalen welke commands gecontroleerd moeten worden? Bv door een array van class names te injecteren, of door een generieke interface op de commands te zetten die hier dan afgecheckt wordt vooraleer de permissie wordt gecontroleerd.